components/patient-measurements/api/src/test/java/org/phenotips/measurements/internal/MeasurementAgeUpdaterTest.java

```diff
@@ -0,0 +1,138 @@
+            new MockitoComponentMockingRule<>(MeasurementAgeUpdater.class);
+
+    @Mock
+    private Object data;
+
+    @Mock
+    public XWikiDocument source;
+
+    @Mock
+    private Event event;
+
+    @Mock
+    private BaseObject patientRecordObj;
+
+    @Mock
+    private BaseObject measurement;
+
+    private static final String DATE_PROPERTY_NAME = "date";
+
+    private static final String AGE_PROPERTY_NAME = "age";
+
+    private List<BaseObject> objects;
+
+    @Before
+    public void setUp() throws NoSuchFieldException, IllegalAccessException
+    {
+        MockitoAnnotations.initMocks(this);
+        when(this.source.getXObject((EntityReference) Matchers.any())).thenReturn(this.patientRecordObj);
+
+        objects = new LinkedList<>();
+        objects.add(measurement);
+        when(this.source.getXObjects((EntityReference)Matchers.any())).thenReturn(objects);
+
+        when(this.measurement.getStringValue(Matchers.anyString())).thenReturn(null);
+    }
+
+    @Test
+    public void testBirthDateNull() throws ComponentLookupException
+    {
+        when(this.measurement.getDateValue(Matchers.anyString())).thenReturn(new Date());
+        when(this.patientRecordObj.getDateValue(Matchers.matches("date_of_birth"))).thenReturn(null);
+
+        this.mocker.getComponentUnderTest().onEvent(event, source, data);
+        verify(this.measurement).removeField(AGE_PROPERTY_NAME);
+    }
+
+    @Test
+    public void testMeasurementDateNull() throws ComponentLookupException
+    {
+        when(this.patientRecordObj.getDateValue(Matchers.matches("date_of_birth"))).thenReturn(new Date());
+
+        this.mocker.getComponentUnderTest().onEvent(event, source, data);
+        verify(this.measurement).removeField(AGE_PROPERTY_NAME);
+    }
+
+    @Test
+    public void testMeasurementTypeBirth() throws ComponentLookupException
+    {
+        when(this.measurement.getStringValue("type")).thenReturn("birth");
+
+        this.mocker.getComponentUnderTest().onEvent(event, source, data);
+
+        verify(this.measurement).setFloatValue(AGE_PROPERTY_NAME, 0);
+        verify(this.measurement).removeField(DATE_PROPERTY_NAME);
+    }
+
+    @Test
+    public void testDateDifference() throws ParseException, ComponentLookupException
+    {
+        SimpleDateFormat dateFormat = new SimpleDateFormat("dd-MM-yyyy hh:mm:ss");
+        String birthDateString = "01-01-2015 12:00:00";
+        String measureDateString = "01-01-2016 12:00:00";
+        Date birthDate = dateFormat.parse(birthDateString);
+        Date measureDate = dateFormat.parse(measureDateString);
+
+        when(this.patientRecordObj.getDateValue(Matchers.matches("date_of_birth"))).thenReturn(birthDate);
+        when(this.measurement.getDateValue(DATE_PROPERTY_NAME)).thenReturn(measureDate);
+
+        this.mocker.getComponentUnderTest().onEvent(event, source, data);
+
+        verify(this.measurement).setFloatValue(eq(AGE_PROPERTY_NAME), eq(365/30.4375f));
+    }
```

Comment:
To continue on this, do that and verify that certain methods are called multiple times with the correct arguments.